perun-tasks-lib/src/main/java/cz/metacentrum/perun/taskslib/dao/jdbc/TaskDaoJdbc.java

```diff
@@ -115,7 +119,7 @@ public int scheduleNewTask(Task task, int engineID) throws InternalErrorExceptio
 			newTaskId = Utils.getNewId(this.getJdbcTemplate(), "tasks_id_seq");
 			this.getJdbcTemplate().update(
 					"insert into tasks(id, exec_service_id, facility_id, schedule, recurrence, delay, status, engine_id) values (?,?,?, " + Compatibility.toDate("?","'DD-MM-YYYY HH24:MI:SS'") + ",?,?,?,?)",
-					newTaskId, task.getExecServiceId(), task.getFacilityId(), getDateFormatter().format(task.getSchedule()), task.getRecurrence(), task.getDelay(), task.getStatus().toString(), engineID);
+					newTaskId, task.getExecServiceId(), task.getFacilityId(), getDateFormatter().format(task.getSchedule()), task.getRecurrence(), task.getDelay(), task.getStatus().toString(), engineID < 0 ? "null" : engineID);
```

Comment:
You can't store string ``"null"`` in ``integer`` type column. It doesn't correspond with your change in DB, where ``engine_id`` can be null.